.github/workflows/gh-linux_mac.yml

```diff
@@ -0,0 +1,360 @@
+          fi;
+          if [[ "${{ matrix.TILEDB_AZURE }}" == "ON" ]]; then
+            bootstrap_args="${bootstrap_args} --enable-azure";
+          fi;
+          if [[ "${{ matrix.TILEDB_GCS }}" == "ON" ]]; then
+            bootstrap_args="${bootstrap_args} --enable-gcs";
+          fi;
+          if [[ "${{ matrix.TILEDB_MEMFS }}" == "ON" ]]; then
+            bootstrap_args="${bootstrap_args} --enable-memfs";
+          fi;
+          if [[ "$TILEDB_TBB" == "OFF" ]]; then
+            bootstrap_args="${bootstrap_args} --disable-tbb";
+          fi
+          if [[ "${{ matrix.TILEDB_TOOLS }}" == "ON" ]]; then
+            bootstrap_args="${bootstrap_args} --enable-tools";
+          fi
+          if [[ "$TILEDB_DEBUG" == "ON" ]]; then
+            bootstrap_args="${bootstrap_args} --enable-debug";
+          fi
+          if [[ "${{ matrix.TILEDB_CI_ASAN }}" == "ON" ]]; then
+            # Add address sanitizer flag if necessary
+            bootstrap_args="${bootstrap_args} --enable-sanitizer=address --enable-debug";
+          fi
+          if [[ "$TILEDB_CI_TSAN" == "ON" ]]; then
+            # Add thread sanitizer flag if necessary
+            bootstrap_args="${bootstrap_args} --enable-sanitizer=thread --enable-debug";
+          fi
+          if [[ "${{ matrix.TILEDB_SERIALIZATION }}" == "ON" ]]; then
+            # Add serialization flag if necessary
+            bootstrap_args="${bootstrap_args} --enable-serialization";
+          fi
+          if [[ "$TILEDB_FORCE_BUILD_DEPS" == "ON" ]]; then
+            # Add superbuild flag
+            bootstrap_args="${bootstrap_args} --force-build-all-deps";
+          fi
+
+          # name: 'Install dependencies'
+
+          mkdir -p $GITHUB_WORKSPACE/build
+          cd $GITHUB_WORKSPACE/build
+
+          # Configure and build TileDB
+          echo "Bootstrapping with '$bootstrap_args'"
+          $GITHUB_WORKSPACE/bootstrap $bootstrap_args
+
+          make -j4
+          make examples -j4
+          make -C tiledb install
+
+          #- run: |
+          cd $GITHUB_WORKSPACE/build
+          ls -la
+
+          if [[ ( ${{ runner.os }} == "Linux" && "${{ matrix.TILEDB_S3 }}" == "ON" ) ]]; then
+            # make sure docker is still running...
+            printenv
+            docker ps -a
+          fi
+
+          make -j4 -C tiledb tiledb_unit
+
+          if [[ "${{ matrix.TILEDB_CI_ASAN }}" == "ON" ]]; then
+            export ASAN_OPTIONS=detect_leaks=0 LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libasan.so.4
+          fi
+
+          if [[ ( "${{ runner.os }}" == "macOS" && "${{ matrix.TILEDB_GCS }}" == "ON" )]]; then
+            # GCS unit tests are temporarily unsupported on CI. Fake success with
+            # this echo.
+            echo ::set-output name=TILEDB_CI_SUCCESS::1
+          else
+            # Bypass Catch2 Framework stdout interception with awk on test output
+            make check | awk '/1: ::set-output/{sub(/.*1: /, ""); print; next} 1'
```

Comment:
We recently switched to running `tiledb_unit -d yes` directly to avoid cmake catching segfaults (which will also avoid the output suppression I think -- though I'm surprised b/c I don't think that was happening on azp w/ `make check`?).

See https://github.com/TileDB-Inc/TileDB/pull/2069/files#diff-b532d55452b891356e4978ba24845f47d4c628fd67bf4a9f4a37389c2f81a888R195 and https://github.com/TileDB-Inc/TileDB/pull/2084

Reply:
Indeed w/ azp there was no problem with the command `##vso[task.setvariable variable=TILEDB_CI_SUCCESS]1`, but after many trials and errors in GHA, `awk` seemed the only possible way to avoid the stdout interception.

Reply:
IIUC we need to run `./tiledb_unit -d yes` directly in order for the new segfault/backtrace handling in #2069 to work, otherwise cmake traps any segfault.

Separately, regarding `stdout`, awk is fine if it still doesn't work with `tiledb_unit` directly, but please add a note that the issue is probably due to the fact that GHA doesn't present as a TTY (not sure why..): https://github.com/actions/runner/issues/241